src/Features/Core/Portable/ConvertToInterpolatedString/ConvertRegularStringToInterpolatedStringRefactoringProvider.cs

```diff
@@ -0,0 +1,118 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.ConvertToInterpolatedString
+{
+    /// <summary>
+    /// Code refactoring that converts a regular string containing braces to an interpolated string
+    /// </summary>
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, LanguageNames.VisualBasic, Name = PredefinedCodeRefactoringProviderNames.ConvertToInterpolatedString), Shared]
+    internal sealed class ConvertRegularStringToInterpolatedStringRefactoringProvider : CodeRefactoringProvider
+    {
+        [ImportingConstructor]
+        [SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
+        public ConvertRegularStringToInterpolatedStringRefactoringProvider()
+        {
+        }
+
+        public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            var (document, _, cancellationToken) = context;
+
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            if (root is null)
+                return;
+
+            var token = root.FindToken(context.Span.Start);
+            if (!context.Span.IntersectsWith(token.Span))
+                return;
+
+            var syntaxKinds = document.GetRequiredLanguageService<ISyntaxKindsService>();
+            if (token.RawKind != syntaxKinds.StringLiteralToken)
+                return;
+
+            var literalExpression = token.Parent;
+            if (literalExpression == null || literalExpression.GetDiagnostics().Any(d => d.Severity == DiagnosticSeverity.Error))
+                return;
```

Comment:
```suggestion
            var literalExpression = token.GetRequiredParent();
            
            // Check the string literal for errors.  This will ensure that we do not try to fixup an incomplete string.
            if (literalExpression.GetDiagnostics().Any(d => d.Severity == DiagnosticSeverity.Error))
                return;
```